packages/gatsby/src/commands/clean.js

```diff
@@ -0,0 +1,50 @@
+const fs = require(`fs-extra`)
+const execa = require(`execa`)
+const path = require(`path`)
+
+const getInfo = require(`./info`)
+
+const YARN_COMMAND = `yarnpkg`
+
+const spawn = cmd => {
+  const [file, ...args] = cmd.split(/\s+/)
+  return execa(file, args, { stdio: `inherit` })
+}
+
+// Executes `npm install` or `yarn install` in rootPath.
+const install = async ({ directory: rootPath, report, useYarn }) => {
+  const prevDir = process.cwd()
+
+  report.info(`Installing packages...`)
+  process.chdir(rootPath)
+
+  try {
+    let cmd = useYarn ? spawn(YARN_COMMAND) : spawn(`npm install`)
```

Comment:
is it worth abstracting spawn as a function? (calling execa directly seems less complex)
```
let cmd = useYarn ? execa(YARN_COMMAND) : execa(`npm`, [`install``])
```